packages/components/src/components/Switch/Switch.stories.tsx

```diff
@@ -0,0 +1,91 @@
+import type { ComponentMeta, ComponentStory } from "@storybook/react";
+import { useForm, Controller, SubmitHandler } from "react-hook-form";
+import React from "react";
+import { Box } from "../../primitives/Box";
+import { Button } from "../Button";
+import { Switch } from "./Switch";
+
+export default {
+  component: Switch,
+  title: "Components/Switch",
+} as ComponentMeta<typeof Switch>;
+
+const Template: ComponentStory<typeof Switch> = (args) => <Switch {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
+  children: "Notifications",
+};
+
+export const Checked = Template.bind({});
+Checked.args = {
+  children: "Enable option",
+  defaultChecked: true,
+  name: "test",
+};
+
+export const Disabled = Template.bind({});
+Disabled.args = {
+  children: "Disabled option",
+  disabled: true,
+};
+
+export const DisabledChecked = Template.bind({});
+DisabledChecked.args = {
+  children: "Disabled option",
+  defaultChecked: true,
+  disabled: true,
+};
+
+export const WithHelpText = Template.bind({});
+WithHelpText.args = {
+  children: "Notifications",
+  helpText: "This enables the notifications settings.",
+};
+
+export const Controlled = (): JSX.Element => {
+  const [checked, setChecked] = React.useState(false);
+
+  return (
+    <Switch
+      checked={checked}
+      onCheckedChange={() =>
+        setChecked((prevIsChecked) => (prevIsChecked === true ? false : true))
+      }
+    >
+      Enable forms
+    </Switch>
+  );
+};
+
+interface FormInputs {
+  enabled: string;
+}
+
+const onSubmit: SubmitHandler<FormInputs> = (data) =>
+  alert(JSON.stringify(data, null, 2));
+
+export const WithReactHookForm = (): JSX.Element => {
+  const { control, handleSubmit } = useForm<FormInputs>({
+    defaultValues: { enabled: "false" },
+  });
+
+  return (
+    <Box.form onSubmit={handleSubmit(onSubmit)}>
+      <Controller
+        control={control}
+        name="enabled"
+        render={({ field }) => (
+          <Switch {...field} onCheckedChange={field.onChange}>
+            Enable forms
+          </Switch>
+        )}
+      />
```

Comment:
Not for this PR but thoughts on a ControlledSwitch (or other name) that already has it wrapped in the Controller so that developers don't need to wrap it in Controller each time or more likely, create a component that's wrapped in Controller individually in the backend and talent repos?

Reply:
Yeah. I've been thinking about that recently. I wonder if we should just wrap our own version of react-hook-form components so they be more easily used in the different applications. Easier and consistent.